codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -375,8 +375,10 @@ private void sendPreface(ChannelHandlerContext ctx) {
                     ChannelFutureListener.CLOSE_ON_FAILURE);
 
             if (isClient) {
-                ctx.fireUserEventTriggered(
-                        Http2ConnectionPrefaceAndSettingsFrameWrittenEvent.INSTANCE);
+                // If this handled is extended by the user and we directly fire the userEvent from this context then
+                // the user will not see the event. We should fire the event through the entire pipeline so it is seen
```

Comment:
also this comment is not correct as we not fire it through the entire pipeline ;) Please adjust